Enable OCI mediatypes

sajayantony opened the pull request:
Signed-off-by: Sajay Antony <sajaya@microsoft.com>

sajayantony committed:
Enable OCI mediatypes

Signed-off-by: Sajay Antony <sajaya@microsoft.com>

adewaleo commented on src/WebManager/Azure.ContainerRegistry/MediaTypeHelper.cs:
```diff
@@ -0,0 +1,22 @@
+{
+    static class MediaTypeHelper
+    {
+        static string _acceptMediaTypes;
```
I wonder if `GetManifestAcceptMediaTypes()` could be made simpler by  making _acceptMediaTypes a Property...

```suggestion
        static string _acceptMediaTypes => string.Join(';',
                    OCI.MediaTypes.IMAGE_INDEX,
                    OCI.MediaTypes.IMAGE_MANIFEST,
                    Docker.MediaTypes.IMAGE_INDEX,
                    Docker.MediaTypes.IMAGE_MANIFEST
                );
        public static string GetManifestAcceptMediaTypes()
        {
            return _acceptMediaTypes;
        }                      
```

or 

```
        public static string GetManifestAcceptMediaTypes() =>  string.Join(';',
                    OCI.MediaTypes.IMAGE_INDEX,
                    OCI.MediaTypes.IMAGE_MANIFEST,
                    Docker.MediaTypes.IMAGE_INDEX,
                    Docker.MediaTypes.IMAGE_MANIFEST
                );
```

See[ expression-bodied members](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members).

adewaleo replied:
This could also be a public property itself instead of a method...

sajayantony replied:
I don't think expression bodied members are able to cache the value.

sajayantony replied:
Fixed in some form.

adewaleo commented on src/WebManager/Azure.ContainerRegistry/MediaTypeHelper.cs:
```diff
@@ -0,0 +1,22 @@
+namespace Azure.ContainerRegistry
```
Could this be the following instead?
```suggestion
namespace WebManager.Utility
```

sajayantony replied:
Fixed

adewaleo commented on src/WebManager/Azure.ContainerRegistry/Docker/MediaTypes.cs:
```diff
@@ -0,0 +1,8 @@
+namespace Azure.ContainerRegistry.Docker
```
Could this be the following instead?
```suggestion
namespace WebManager.Utility.Docker
```

adewaleo reviewed (commented):
Some comments @sajayantony

sajayantony commented:
Fixed namespaces and moved content around. Also fixed a bug to get the correct manifest as well. 
I would like get Gil's change checked in following this.

adewaleo reviewed (approved):
Not sure what the bug fix was but this looks good to me.
